src/utils/configMerge.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { isObject } from '@adobe/spacecat-shared-utils';
+
+function isDuplicateJob(job1, job2) {
+  return job1.group === job2.group && job1.type === job2.type;
+}
+
+function mergeJobsArray(targetJobs, sourceJobs) {
+  const mergedJobs = [...targetJobs];
+
+  sourceJobs.forEach((sourceJob) => {
+    const duplicateIndex = mergedJobs.findIndex(
```

Comment:
Using `findIndex` inside a for-loop creates a complexity of `O(n)`. We could avoid that by using a Map instead. Composing a key out of the group and type, it would override duplicates inside the Map, reducing the complexity to `O(1)`
Example:
```
function mergeJobsArray(targetJobs, sourceJobs) {
  // Create a Map from targetJobs for efficient lookups
  const jobsMap = new Map(targetJobs.map(job => [`${job.group}-${job.type}`, job]));

  // Iterate over sourceJobs to merge or add
  sourceJobs.forEach((sourceJob) => {
    const jobKey = `${sourceJob.group}-${sourceJob.type}`;
    jobsMap.set(jobKey, sourceJob); // This will overwrite if the key exists, or add if it doesn't
  });

  // Convert the Map values back to an array
  return [...jobsMap.values()];
}
```